tool_quality_checks/quality_checks.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+""" TODO Docstring. """
+import os
+import argparse
+import logging
+
+from configparser import RawConfigParser
+from db import ThreediDatabase
+
+logger = logging.getLogger(__name__)
+
+
+def quality_checks(settings):
+    """Overall function for checking our model data"""
+
+    # get database connection
+    db = ThreediDatabase(settings)
+    # TODO: Make it only if install is on.
+    db.initialize_db()
+
+    # get v2_table_names
+    v2_table_names = db.select_table_names("v2%")
+    for table_name in v2_table_names:
+        if db.get_count(table_name) > 0:
+            print(table_name, db.get_count(table_name))
+            db.perform_checks_with_sql(settings, table_name, check_type="completeness")
+            db.perform_checks_with_sql(settings, table_name, check_type="quality")
+
+    db.populate_geometry_columns()
+
+
+def resolve_ini(custom_ini_file):
+    """
+    decide which ini file to use
+    """
+    # get default ini for testing purposes
+    default_ini_relpath = "test\\data\\instellingen_test.ini"
+    default_ini_relpath = os.path.join(os.path.dirname(__file__), default_ini_relpath)
+    if custom_ini_file is None:
+        logger.info(
+            "[*] Using default ini file {}".format(
+                os.path.basename(default_ini_relpath)
+            )
+        )
+        return default_ini_relpath
+    if all((os.path.exists(custom_ini_file), os.path.isfile(custom_ini_file))):
+        logger.info(
+            "[*] Using custom ini file {}".format(os.path.basename(custom_ini_file))
+        )
+        return custom_ini_file
+    else:
+        raise ("Error: Could not find the ini file {}".format(custom_ini_file))
+
+
+class settingsObject(object):
+    """Contains the settings from the ini file"""
+
+    def __init__(self, inifile):
+        config = RawConfigParser()
+        config.read(inifile)
+
+        # inifile
+        self.ini = inifile
+
+        for section in config.sections():
+            for key, value in config.items(section):
+                setattr(self, key, value)
```

Comment:
Hm, dit is erg dynamisch.
Als je later iets aanroept dat niet in de ini stond, dan krijg je een AttributeError. Is dat goed genoeg?

Reply:
Ja klopt. Dit is erg dynamisch, maar ook erg fijn dat je dus niet de hele tijd allerlei dingen moet gaan definieren. Als ik iets toevoeg, hoef ik alleen maar het ergens anders weer op te pakken.

Omtrent de ini-file controle aan de deur heb je wel een punt, daar zouden we samen nog wel even over kunnen nadenken hoe dat mooi op te pakken.